make-zip

```diff
@@ -1,80 +1,183 @@
+    :raise UnableToCreateUsercodeBundleException: An error occurred during creation.
+    """
+    entrypoint_path = usercode_dir / ENTRYPOINT
+
+    # Validate arguments
+    if not usercode_dir.exists() or not usercode_dir.is_dir():
+        raise UnableToCreateUsercodeBundleException(
+            f"Usercode directory ({usercode_dir}) is not a directory or doesn't exist.",
+        )
+
+    if not entrypoint_path.exists():
+        raise UnableToCreateUsercodeBundleException(
+            f"Usercode directory ({usercode_dir}) does not contain a {ENTRYPOINT} file.",
+        )
+
+    if len(team_tla) == 0:
+        raise UnableToCreateUsercodeBundleException("Team TLA cannot be empty.")
+
+    # Create the ZIP
+    with ZipFile(output_file, "w", compression=ZIP_DEFLATED) as zip_file:
+        # Add the usercode
+        for folderName, subfolders, filenames in os.walk(usercode_dir):
+            relativeFolder = os.path.relpath(folderName, usercode_dir)
+            for filename in filenames:
+                relative_path = os.path.join(relativeFolder, filename)
+                real_path = os.path.join(folderName, filename)
+
+                # Check that the file should not be ignored
+                if not is_file_ignored(relative_path):
+                    zip_file.write(real_path, relative_path)
+
+        # Add bundle.toml
+        bundle_data = get_bundle_dict(
+            "0.0.0.0dev",  # This should eventually be fetched from an API
+            f"robot-{team_tla}",
+            wifi_psk,
+            wifi_region,
+        )
+        bundle_str = toml.dumps(bundle_data)
+        zip_file.writestr("bundle.toml", bundle_str)
+
+
+def main() -> None:
+    """Pack a robot.zip from user code."""
+    parser = argparse.ArgumentParser(description="Pack a robot.zip from user code.")
+    parser.add_argument(
+        "team_tla",
+        nargs="?",
+        default="SRZ",  # Default to SRZ if no TLA is provided
+        help="The TLA of the team the zip is being created for",
+    )
+    parser.add_argument(
+        "usercode",
+        type=Path,
+        help="user code source directory, containing a robot.py",
+    )
+    parser.add_argument(
+        "output",
+        nargs="?",
+        type=Path,
+        help="output zip file",
+        default="robot.zip",
+    )
+    parser.add_argument(
+        "-p",
+        "--psk",
+        default="beeeeees",
+        help="WiFi PSK used to secure robot hotspot",
+    )
+    parser.add_argument(
+        "-r",
+        "--region",
+        default="GB",
+        help="Region information for the WiFi hotspot",
+    )
+
+    args = parser.parse_args()
+
+    try:
+        create_usercode_bundle(
+            args.team_tla,
+            args.usercode,
+            args.output,
+            args.psk,
+            args.region,
+        )
+    except UnableToCreateUsercodeBundleException as e:
+        print(f"Error: {e}", file=sys.stderr)
```

Comment:
Please ensure that the tool exits with a suitable error exit code in case of errors.
An easy way to do this would be:
```suggestion
    except UnableToCreateUsercodeBundleException as e:
        exit(f"Error: {e}")
```
Alternatively we'll need to pass the exit code back through the `if __name__ == '__main__':` etc.